### PR TITLE
Create component tests for interface between c and c++ for FIMDB

### LINKED ISSUE #11332: Create unit test for new methods to integrate the existing code with the new DB implementation
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

This issue intends to add tests to assure that the new methods is working properly. It implies:

- [x] #11562
- [x] #11563
- [x] #11564

### LINKED ISSUE #11564: Create tests for the new implementation of common methods necessary to manage the database in `db.cpp`.
Hi team!!

The purpose of this issue is to create the needed component tests for the FIM db lib interface. 

The code of this interface is written in C++, but with C headers. These tests should be made using GTest. 

# Needed tests
- [x] `fim_db_init` 
- [x] `fim_run_integrity` 
- [x] `fim_sync_push_msg`

### LINKED ISSUE #11848: Fix bug on fim_run_integrity function inside db.cpp interface
|Wazuh version|Component|
|---|---|
| 4.4.0+| Wazuh FIM |

### Description

I found a bug in **fim_run_integrity** function when execute this function the thread started but there is a crash in some moment. Also I can collect the error message which is:

`AddressSanitizer:DEADLYSIGNAL
=================================================================
==95482==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x000000000000 bp 0x7f45fa0f6970 sp 0x7f45fa0f6938 T25)
==95482==Hint: pc points to the zero page.
==95482==The signal is caused by a READ memory access.
==95482==Hint: address points to the zero page.

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (<unknown module>) 
Thread T25 created by T0 here:
    #0 0x7f46034e2805 in pthread_create (/usr/lib/x86_64-linux-gnu/libasan.so.5+0x3a805)
    #1 0x7f46033cbe08 in std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) (/home/francorivero/Desktop/Wazuh_repositories/wazuh/src/shared_modules/dbsync/build/lib/libdbsync.so+0x26ce08)
    #2 0x55c9cd71701d in fim_run_integrity /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/src/db.cpp:101
    #3 0x55c9cd69583d in DBTest_TestFimSyncPushMsg_Test::TestBody() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/tests/db/ComponentTest/dbInterface/dbTest.cpp:100
    #4 0x55c9cd7ababf in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2607
    #5 0x55c9cd7a3e26 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2643
    #6 0x55c9cd7797ef in testing::Test::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2682
    #7 0x55c9cd77a231 in testing::TestInfo::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2861
    #8 0x55c9cd77ab2e in testing::TestSuite::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:3015
    #9 0x55c9cd78a4c0 in testing::internal::UnitTestImpl::RunAllTests() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:5855
    #10 0x55c9cd7acfed in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2607
    #11 0x55c9cd7a5064 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2643
    #12 0x55c9cd788c25 in testing::UnitTest::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:5438
    #13 0x55c9cd715d00 in RUN_ALL_TESTS() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/include/gtest/gtest.h:2490
    #14 0x55c9cd715b5e in main /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/tests/db/ComponentTest/dbInterface/main.cpp:6
    #15 0x7f4602aec0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)

==95482==ABORTING
`

#### Steps to reproduce:

1. Delete "DISABLED_" before test name 
2. Compile wazuh with tests flag
3. Execute component tests

#### Actual behavior:
The TestFimSyncPushMsg, TestFimRunIntegrity test is failing

#### Expected behavior:
The TestFimSyncPushMsg, TestFimRunIntegrity  must pass

#### Definition of done:
- Test TestFimRunIntegrity passed
- Delete `DISABLED_` tag in TestFimRunIntegrity

### LINKED ISSUE #11850: Fix bug on fim_db_transaction_sync_row function inside db.cpp interface
|Wazuh version|Component|
|---|---|
| 4.4.0+| Wazuh FIM |

### Description

I found a bug in **fim_db_transaction_sync_row** function when execute this function the null entry is not manage in the method and generate an error. I can collect the error message which is:

`AddressSanitizer:DEADLYSIGNAL
=================================================================
==107549==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x5593f5555f35 bp 0x7ffd9f1dc370 sp 0x7ffd9f1dc1e0 T0)
==107549==The signal is caused by a READ memory access.
==107549==Hint: address points to the zero page.
    #0 0x5593f5555f34 in fim_db_transaction_sync_row /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/src/db.cpp:143
    #1 0x5593f54dc0c2 in DBTest_TestSyncRowTransactionWithInvalidFimEntry_Test::TestBody() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/tests/db/ComponentTest/dbInterface/dbTest.cpp:258
    #2 0x5593f55e9b75 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2607
    #3 0x5593f55e1edc in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2643
    #4 0x5593f55b78a5 in testing::Test::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2682
    #5 0x5593f55b82e7 in testing::TestInfo::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2861
    #6 0x5593f55b8be4 in testing::TestSuite::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:3015
    #7 0x5593f55c8576 in testing::internal::UnitTestImpl::RunAllTests() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:5855
    #8 0x5593f55eb0a3 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2607
    #9 0x5593f55e311a in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:2643
    #10 0x5593f55c6cdb in testing::UnitTest::Run() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/src/gtest.cc:5438
    #11 0x5593f5553db6 in RUN_ALL_TESTS() /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/external/googletest/googletest/include/gtest/gtest.h:2490
    #12 0x5593f5553c14 in main /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/tests/db/ComponentTest/dbInterface/main.cpp:6
    #13 0x7f04008d70b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #14 0x5593f54d12bd in _start (/home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/build/bin/fim_db_interface_test+0x2d12bd)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/francorivero/Desktop/Wazuh_repositories/wazuh/src/syscheckd/db/src/db.cpp:143 in fim_db_transaction_sync_row
==107549==ABORTING
`

#### Steps to reproduce:

1. Delete "DISABLED_" before test name 
2. Compile wazuh with tests flag
3. Execute component tests

#### Actual behavior:
The TestSyncRowTransactionWithInvalidFimEntry test is failing

#### Expected behavior:
The TestSyncRowTransactionWithInvalidFimEntry  must pass

The input which fail is `fim_db_transaction_sync_row(handler, nullptr);`

####Definition of Done
- Delete `DISABLED_` tag for tests
- TestSyncRowTransactionWithInvalidFimEntry must pass correctly

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

I created component tests to db.cpp interface

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Unit tests passed
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] cppcheck (static code analysis)
  - [x] Coverage (unit tests coverage code)
image
  - [x] Sformat (code style)
  - [x] Scan build
  
image
image

image

Closes #11564
Closes #11332
Closes #11850
Closes #11848

### CODE REVIEW

File: src/syscheckd/src/db/src/db.cpp [hunk 1]
```diff
@@ -100,6 +100,11 @@ DBSYNC_HANDLE DB::DBSyncHandle()
     return FIMDB::instance().DBSyncHandle();
 }
 
+void DB::registerRSync()
```
Comment (Dwordcito) [inline src/syscheckd/src/db/src/db.cpp:103]: Why are we exposing this? In the RunIntegrity the registration is already taking place. We are giving the caller responsibility for something that he can do implicitly.

FIMDB::runIntegrity()

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/syscheckd/src/db/src/db.cpp (hunk 1 above)
Comment (FrancoRivero) [reply src/syscheckd/src/db/src/db.cpp:103]: Yes, I didn't mind in that ! I have to add a mock to this call !

File: src/syscheckd/src/db/src/db.cpp [hunk 2]
```diff
@@ -197,33 +215,37 @@ TXN_HANDLE fim_db_transaction_start(const char* table, result_callback_t row_cal
...
-    std::unique_ptr<DBItem> syncItem;
-    auto retval { FIMDB_OK };
+    auto retval { FIMDB_ERR };
 
-    if (entry->type == FIM_TYPE_FILE)
-    {
-        syncItem = std::make_unique<FileItem>(entry);
-    }
-    else
+    if (entry)
```
Comment (antoniomanuelfr) [inline src/syscheckd/src/db/src/db.cpp:220]: We can avoid an indentation level
```suggestion
    if (entry)
    {
        return FIMDB_ERR;
     }
     std::unique_ptr<DBItem> syncItem;
```

**Review by antoniomanuelfr — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by jotacarma90 — APPROVED** [review]
LGTM

```
<syscheckd>=================== Running Tests       ===================<syscheckd>
[registrykey_unit_test: PASSED]
[fim_file_interface_test: PASSED]
[fimdb_unit_test: PASSED]
[registryvalue_unit_test: PASSED]
[fileitem_unit_test: PASSED]
[fim_db_interface_test: PASSED]
```

**Review by antoniomanuelfr — APPROVED** [review]

**Pull request merged**